### PR TITLE
reconcile services

### PR DESCRIPTION
**Pull request by calvix**
As the annotation is now needed for guest cluster monitoring we need to be able to update old cluster and the that means enable reconciliation of the service to desired state

Also since the etcd port is different across providers and it needs to be part of annotation

### CODE REVIEW

File: service/controller/v3/resource/service/resource.go [hunk 1]
```diff
@@ -44,6 +45,32 @@ func (r *Resource) Name() string {
...
+	for _, l := range list {
+		if l.Name == name {
+			return l, nil
+		}
+	}
+
+	return nil, microerror.Mask(notFoundError)
+}
+
+func isServiceModified(a, b *corev1.Service) bool {
```
Comment (xh3b4sd) [inline service/controller/v3/resource/service/resource.go]: We definitely need unit tests for this.

File: service/controller/v3/resource/service/update.go [hunk 2]
```diff
@@ -3,11 +3,29 @@ package service
...
+
+		_, err := r.k8sClient.CoreV1().Services(serviceToUpdate.Namespace).Update(serviceToUpdate)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "updated services")
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "no need to update services")
+	}
```
Comment (xh3b4sd) [inline service/controller/v3/resource/service/update.go]: Well, why wasn't this already implemented? Anything else we do not properly reconcile? @kopiczko @TheoBrigitte

Comment (kopiczko) [reply service/controller/v3/resource/service/update.go]: It was some copy paste by mr @JosephSalisbury

File: service/controller/v3/resource/service/resource.go (hunk 1 above)
Comment (calvix) [reply service/controller/v3/resource/service/resource.go]: ack

File: service/controller/v3/resource/service/update.go (hunk 2 above)
Comment (calvix) [reply service/controller/v3/resource/service/update.go]: yeah its same in all operators, i already have prepared PRs there a well

**Comment by calvix** [comment]
RFR, tested  and it is looking fine

File: service/controller/v3/resource/service/resource_test.go [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+							Protocol:   corev1.ProtocolTCP,
+							Port:       httpsPort,
+							TargetPort: intstr.FromInt(httpsPort),
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "case 1: wrong type",
```
Comment (kopiczko) [inline service/controller/v3/resource/service/resource_test.go]: Please add this is because a pointer is expected. It took mea while to realize.

**Pull request merged**